src/bundle/Controller/SearchController.php

```diff
@@ -0,0 +1,142 @@
+use Symfony\Component\HttpFoundation\Request;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use Pagerfanta\Pagerfanta;
+use eZ\Publish\Core\Pagination\Pagerfanta\ContentSearchAdapter;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory;
+use EzSystems\EzPlatformAdminUi\Form\SubmitHandler;
+
+class SearchController extends Controller
+{
+    /** @var SearchService */
+    private $searchService;
+
+    /** @var PagerContentToDataMapper */
+    protected $pagerContentToDataMapper;
+
+    /** @var Router */
+    protected $router;
+
+    /** @var FormFactory */
+    private $formFactory;
+
+    /** @var SubmitHandler */
+    private $submitHandler;
+
+    /**
+     * @param SearchService $searchService
+     * @param PagerContentToDataMapper $pagerContentToDataMapper
+     * @param RouterInterface $router
+     * @param FormFactory $formFactory
+     * @param SubmitHandler $submitHandler
+     */
+    public function __construct(
+        SearchService $searchService,
+        PagerContentToDataMapper $pagerContentToDataMapper,
+        RouterInterface $router,
+        FormFactory $formFactory,
+        SubmitHandler $submitHandler
+    ) {
+        $this->searchService = $searchService;
+        $this->pagerContentToDataMapper = $pagerContentToDataMapper;
+        $this->router = $router;
+        $this->formFactory = $formFactory;
+        $this->submitHandler = $submitHandler;
+    }
+
+    /**
+     * Renders the simple search form and search results.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function searchAction(Request $request): Response
+    {
+        $search = $request->query->get('search');
+        $limit = $search['limit'] ?? 10;
+        $page = $search['page'] ?? 1;
+        $query = $search['query'];
+
+        $form = $this->formFactory->createSearchForm(new SearchData($limit, $page, $query), 'search', [
+            'method' => 'GET',
+            'csrf_protection' => false,
+        ]);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $result = $this->submitHandler->handle($form, function (SearchData $data) use ($form) {
+                $limit = $data->getLimit();
+                $page = $data->getPage();
+                $queryString = $data->getQuery();
+
+                $query = new Query();
+                $query->filter = new Criterion\LogicalAnd(
+                    [
+                        new Criterion\Visibility(Criterion\Visibility::VISIBLE),
+                        new Criterion\FullText($queryString),
+                    ]
+                );
+
+                $query->limit = $limit;
+                $query->offset = ($page - 1) * $limit;
```

Comment:
You don't need to set `limit` and `offset` manually, pagerfanta should take care of this.

Reply:
Ok, so apparently this already works with `ContentSearchAdapter` from kernel (which I didn't know or even expect to be implemented there, sorry). Not 100% convinced this is clean, as being PAPI developer I got confused, but seems like a way to go.

Reply:
Ok, so I will remove `limit` and `offset` for query

Reply:
yep, sorry for the mess.